### ISSUE TITLE
Failed to register service worker behind reverse proxy and subdirectory

### ISSUE THREAD

**Issue by ZhangZisu** [cdr/code-server#2106]
- Web Browser: Chrome/85.0.4183.106
- Local OS: Microsoft Windows [Version 10.0.19041.508]
- Remote OS: Linux 5.8.8-arch1-1
- Remote Architecture: x86-64
- `code-server --version`: 3.5.0 de41646fc402b968ca6d555fdf2da7de9554d28a

Chrome Dev Tools output:
```
The path of the provided scope ('/code') is not under the max scope allowed (set by Service-Worker-Allowed: '/code/'). Adjust the scope, move the Service Worker script, or use the Service-Worker-Allowed HTTP header to allow the scope.
```

reverse proxy definition in nginx.conf:
```
location /code/ {
            proxy_pass http://localhost:8003/;
            proxy_set_header Host $host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection upgrade;
            proxy_set_header Accept-Encoding gzip;
            # proxy_hide_header Service-Worker-Allowed;
            # add_header Service-Worker-Allowed /code;
}
```

However, uncomment these two lines worked for me:
```
            # proxy_hide_header Service-Worker-Allowed;
            # add_header Service-Worker-Allowed /code;
```

**Comment by ZhangZisu** [comment]
My code-server/config.yaml is:
```
bind-addr: 127.0.0.1:8003
auth: none
cert: false
```

**Comment by ZhangZisu** [comment]
Screenshot:
![image](https://user-images.githubusercontent.com/21094314/93485961-39b5fe00-f936-11ea-9a3a-7a6162a149d2.png)

**Comment by code-asher** [comment]
Thanks for the report! This will be fixed in 3.5.1. Duplicate of https://github.com/cdr/code-server/issues/2076

**Comment by Rishab1207** [comment]
Hi @thezzisu, were you able to resolve your issue? If yes, then can you please tell me that was this reverse proxy approach successful in registering PWA?
